core/pkg/service/flag-evaluation/flag_evaluator.go

```diff
@@ -269,6 +259,42 @@ func (s *FlagEvaluationService) ResolveObject(
 	return res, err
 }
 
+// resolve is a generic flag resolver
+func resolve[T constraints](
+	ctx context.Context,
+	logger *logger.Logger,
+	resolver func(context context.Context, reqID, flagKey string, ctx *structpb.Struct) (T, string, string, error),
+	flagKey string,
+	evaluationContext *structpb.Struct,
+	resp response[T],
+	metrics *telemetry.MetricsRecorder,
+) error {
+	reqID := xid.New().String()
+	defer logger.ClearFields(reqID)
+
```

Comment:
as a follow-up, we could do a bit of refactoring to the telemetry pkg to expose a utility function to create the semantic convention for FF instead of having that embedded into the metric part. 
This way we can add an event to the span when the evaluation is performed as the semantic conventions suggest ([docs](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/feature-flags.md#convention)).

Reply:
I hope you are referring to this section,

`A flag evaluation SHOULD be recorded as an Event on the span during which it occurred.`

Isn't this highlighting the feature flag evaluation side (client sdk or server sdk) of the evaluation? If you can explain this a bit, I can follow up on the implementation :)

Reply:
Yes, this is the section I am referring to :) 
Currently, while recording the metric we add as a dimension the three attributes `provider`, `key`, and `variant`. We could refactor that piece of code to expose the creation of these attributes and then in this piece of code we could add the following after the call to `resolver`:
```go
span := trace.SpanFromContext(ctx)
span.SetAttributes(telemetry.FFAttributes(flagKey, variant))
```
`SpanFromContext` returns the current active Span in the context. If no Span is active in the context, a noop span implementation is returned. `telemetry.FFAttributes(string,string)` is the function that builds the attribute for the FF semantic convention. 
I hope this clarifies a bit more. Let me know if something is still unclear :)

Reply:
Thanks for the explanation. Yes, it makes sense to add span attributes as you suggest. 

Will refactor and add this with the next commit.

Reply:
@thisthat please check the proposed improvement [1] 

And here is the span with attributes 

![image](https://user-images.githubusercontent.com/8186721/231521211-3e8e687e-d06a-4c3f-9bfa-dd008fdc54a7.png)

[1] - https://github.com/open-feature/flagd/blob/8eb1402b15a02230fdd9a8e4dd9e8fe681aec267/core/pkg/telemetry/utils.go